src/Core/Form/IdentifiableObject/CommandBuilder/Product/CustomizationFieldsCommandBuilder.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Form\IdentifiableObject\CommandBuilder\Product;
+
+use PrestaShop\PrestaShop\Core\Domain\Product\Customization\Command\RemoveAllCustomizationFieldsFromProductCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\Customization\Command\SetProductCustomizationFieldsCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
+
+/**
+ * Builds commands from product customizations form
+ */
+final class CustomizationFieldsCommandBuilder implements ProductCommandBuilderInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function buildCommand(ProductId $productId, array $formData): array
+    {
+        if (!isset($formData['customizations'])) {
+            return [];
+        }
+
+        $customizations = $formData['customizations'];
+
+        if (empty($customizations) || !isset($customizations['customization_fields'])) {
+            return [new RemoveAllCustomizationFieldsFromProductCommand($productId->getValue())];
+        }
+
+        return [
+            new SetProductCustomizationFieldsCommand(
+                $productId->getValue(),
+                $this->buildCustomizationFields($customizations['customization_fields'])
+            ),
+        ];
+    }
+
+    /**
+     * @param array $customizationsFormData
+     *
+     * @return array<int, array<string, mixed>>
+     */
+    private function buildCustomizationFields(array $customizationsFormData): array
+    {
+        $customizationFields = [];
+        foreach ($customizationsFormData as $customization) {
+            $customizationFields[] = [
+                'type' => (int) $customization['type'],
+                'localized_names' => $customization['name'],
+                'is_required' => (bool) $customization['required'],
+                'added_by_module' => false,
```

Comment:
Is it `false` because we know it's never `true` or is it a default value because we have no idea how and why this field is supposed to be set to `true` (as we discussed) ?

Reply:
its kind of a default value, because we are not adding it by module :man_shrugging: 
It is not exactly clear how this field is used, but it seems self explanatory

Reply:
https://www.prestashop.com/forums/topic/549666-guide-the-new-customization-system-in-prestashop-17/
I know only that:
```
A new column was added: `is_module` which means that the field belongs to a module
When this column has a value of 1, the field won't be displayed on the front
Make sure to set the value to 1 if your module creates a custom field for the product
Your module can set the value of the input `#product_customization_id` to include the customization in the cart
Ideal scenario:
- The client makes a customization and saves it via ajax
- Your module returns an `id_customization` and puts it in `#product_customization_id`
- Your module clicks the add to cart button
```

Reply:
Interesting, this means the customization field from modules should not be displayed in FO
Then if we want to avoid modifying them we should hide them in the BO as well, or at least it's the safest for now This way we only handle core fields not module's ones in the BO

If some modules relied on the fact that they could edit their fields in the product page we can always add them back, but we need to make sure that we don't forcibly set `is_module` to false when editing them This can be done after this PR though and needs to be confirmed by @PrestaShop/product-team @MatShir